First attempt to cache PLT

lasseebert opened the pull request:
Adds a cache of the entire _build dir to save PLT for next build.

Uses Elixir and Erlang versions as part of the cache key, since we want to rebuild PLT if they change.

skovmand commented:
@lasseebert I almost started doing this one too. I saw [a question on elixirforum](https://elixirforum.com/t/caching-dialyzer-output-with-github-actions/27648/5) where a guy couldn't get the PLT caching to work on GitHub Actions. Apparently he had to use another path for the PLT than `_build`. So maybe that could help you if there's problems.

lasseebert commented:
@skovmand Thanks :crossed_fingers:

lasseebert commented:
@skovmand: I think it is there now :)

Added `actions/cache@v1` step that caches the entire _build dir with a cache key that consists of the build-matrix params and a hash of mix.lock.

If it can't find the full key, it will look up any key with correct build-matrix (but only in own branch or parent branches, since that is how caching works in github actions)

I added steps to force compile, otherwise the dialyzer step would report an error about Dialyzer mix task module already exists when trying to compile before dialyzing.

lasseebert commented:
@skovmand: Yes, it feels a bit strange. I had trouble without `--force`, which I think is because the cache is restored with compiled files (we cache the entire _build dir), but checkout makes the code files have newer timestamps, which makes the compiler think they should be recompiled.

Somewhere in that, it errors if code is not force-compiled.
